cclib/method/ddec.py

```diff
@@ -221,23 +280,19 @@ def calculate_refcharges(self):
         stockholder_bigW = numpy.sum(stockholder_w, axis=0)
         localized_bigW = numpy.sum(localized_w, axis=0)
 
-        self.logger.info("Creating first reference charges.")
-        self.refcharges = numpy.zeros((atoms))
-        self._localizedcharges = numpy.zeros((atoms))
-        self._stockholdercharges = numpy.zeros((atoms))
+        refcharges = numpy.zeros((atoms))
+        localizedcharges = numpy.zeros((atoms))
+        stockholdercharges = numpy.zeros((atoms))
 
         for atomi in range(atoms):
-            chargedensity = copy.deepcopy(self.chgdensity)
-            localized_weight = localized_w[atomi] / localized_bigW
-            chargedensity.data *= localized_weight
-            self._localizedcharges[atomi] = self.data.atomnos[atomi] - chargedensity.integrate()
-
-            chargedensity = copy.deepcopy(self.chgdensity)
-            stockholder_weight = stockholder_w[atomi] / stockholder_bigW
-            chargedensity.data *= stockholder_weight
-            self._stockholdercharges[atomi] = self.data.atomnos[atomi] - chargedensity.integrate()
+            localizedcharges[atomi] = self.data.atomnos[atomi] - self.chgdensity.integrate(
```

Comment:
I would put a code comment above each line referencing the exact equation number from the paper.

Reply:
I've added the equation numbers above the appropriate lines. Thanks.